eng/scripts/Generate-Dependency-Test-References.ps1

```diff
@@ -0,0 +1,75 @@
+
+.DESCRIPTION
+.NET pipelines will run centain number of test dependencies in each entry of job matrix. 
+The Script is to split all test dependencies into difference sets for later use. 
+
+.PARAMETER ProjectListFilePath
+The file path contains all test dependencies. E.g. The file generated by 'eng/service.proj' in dependecny.tests.yml `Find all test dependencies` step
+
+.PARAMETER ProjectFilesOutputFolder
+The folder for all test dependency split files e.g. `Project_1.props` which are ready to publish. E.g. $(Build.ArtifactStagingDirectory) 
+
+.PARAMETER NumOfTestsPerJob
+The number of test run on each job.
+
+.PARAMETER MatrixConfigsFile
+The file path of the matrix job configuration. E.g. "/eng/pipelines/templates/stages/platform-matrix.json"
+
+.PARAMETER ProjectFileConfigName
+The config name of the project file name. Hardcoded in "\eng\pipelines\templates\jobs\ci.yml" `PreGenerationSteps` for matix generation. E.g. 'ProjectListOverrideFileProperty' 
+
+.EXAMPLE
+Run script with default parameters.
+
+Generate-Dependency-Test-References.ps1 -ProjectFilesOutputFolder "$(Build.ArtifactStagingDirectory)/projects" -ProjectListFilePath "eng/projects.txt" `
+-MatrixConfigsFile "eng/pipelines/templates/stages/platform-matrix.json" -ProjectFileConfigName "ProjectListOverrideFileProperty"
+
+#>
+Param (
+    [string] $ProjectListFilePath,
+    [string] $ProjectFilesOutputFolder,
+    [int] $NumOfTestsPerJob = 20,
+    [string] $MatrixConfigsFile,
+    [string] $ProjectFileConfigName
+)
+
+$packages = Get-Content $ProjectListFilePath
+$numOfGroups = [Math]::Floor([decimal]($packages.Count / $NumOfTestsPerJob ))
+$pkgGroup = [System.Collections.ArrayList]::new()
+$i = 0
+for ($i = 0; $i -lt $numOfGroups; $i++) { 
+  $pkgGroup.Add($packages[( $i * $NumOfTestsPerJob )..(($i + 1) * $NumOfTestsPerJob )])
+}
+if ($packages.Count -gt $numOfGroups * $NumOfTestsPerJob) {
+  $pkgGroup.Add($packages[($numOfGroups * $NumOfTestsPerJob)..$packages.Count])
+}
+
+$null = New-Item -Path $ProjectFilesOutputFolder -ItemType "directory" -Force 
+
+for ($i = 0; $i -lt $numOfGroups; $i++) {
+  $projectFilePath = "$ProjectFilesOutputFolder/Project_$i.props"
+  $null = New-Item -Path $projectFilePath -ItemType "file"
+  # Retain the structure without ProjectReference.
+  $templateXml=[xml]'<Project></Project>'  
+  $projectNode = $templateXml.SelectNodes("/Project")
+  $itemGroupNode=$templateXml.CreateNode("element", "ItemGroup", "")
+  $projectNode.AppendChild($itemGroupNode)
+  foreach($pkg in $pkgGroup[$i]) {
+    $newElem = $templateXml.CreateNode("element", "ProjectReference", "")
+    $newElemAttr = $templateXml.CreateAttribute("Include")
+    $newElemAttr.InnerText = $pkg
+    $newElem.Attributes.Append($newElemAttr)
+    
+    $itemGroupNode.AppendChild($newElem)
+    $templateXml.Save($projectFilePath)
+  }
+}
+
+$platformJson = Get-Content $MatrixConfigsFile | ConvertFrom-Json
+$projectFiles = Get-ChildItem "$ProjectFilesOutputFolder\Project_*.props"
```

Comment:
Do we want to add `Nprops` to the job display name? Might be nice to update the `displayNames` property or add things in the object grouped format (https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/README.md#matrix-config-file-syntax).

Reply:
Yes I mentioned we might want to us groups so we can get a better display name as well. Something like "Dependency Group X" or something like that.